### ISSUE TITLE
Make opening the `/edit` view for items easier

### ISSUE THREAD

**Issue by allcaps** [plone/plonetheme.barceloneta#65]
If I'm in the edit environment `/Plone/folder_contents` and do the following:

1. Click on a folder (goes to `/Plone2/myfolder/folder_contents`)
2. Click on a page in that folder (goes to `/Plone2/myfolder/mypage/view`)

It is weird that a click on a folder does something different than a click on an item. When I'm in the edit environment I want to stay in the edit environment. On the second click I expect to go to the *edit* view at `/Plone2/myfolder/mypage/edit`. 

Alternatives to go to the edit view are: Click gear > click 'Edit' OR click table row > click `Edit` in the toolbar. Both two clicks. One to much.

There is also a shortcut icon on the table row. Which does the same as clicking the row itself. While  icons like these are a prefect opportunity to supply additional functionality. Eg: pencil icon to edit.

I understand that this is the default behaviour of Plone for years but I think it should be easier to navigate to the edit view of non-folerisch content types.

My question: How can I make the table row clicks always open `/folder_contents` and `/edit` for all other items?

![screen shot 2015-12-02 at 11 32 29](https://cloud.githubusercontent.com/assets/1969342/11529359/de698b7e-98ec-11e5-94f0-899ef3511ef9.png)

**Comment by davilima6** [comment]
`folder_contents` is not _edit mode_ as you put it. The page you get to when you click `Edit` button in the context of a folder is what would be correctly called its edit mode. The `Contents` button takes you to a 3rd mode which is _file explorer-like_, including tools for batch operations.

That being said I agree one-click access to item Edit view would be nice to have. It's a CMS so imo primary CMS actions should be always available, as much as possible. During Plone 5 Beta we had a suggestion for an Edit icon but I couldn't find the issue now.

I think that should be done and that we should keep the remaining actions under a Lighting Icon, equal to the one in the toolbar. Then add a separator and we can have "Set as default page" in there.

**Comment by allcaps** [comment]
You are right. Folders have both `folder_contents` and `edit` views. With 'edit' I mean the wider concept 'edit interface' or 'admin interface'.

> ... Lightning Icon, equal to the one in the toolbar.

The current view icon in the listing is an *external link* icon and not an *eye* as in the toolbar.

I see the folder contents listing as a way to navigate to an edit form. I'm always confused when I click a non-folderish item and endup in the view. Which is IMHO a weird detour. Traditionally Plone does in context editing. Than it makes sense to jump to the view. But I think Barceloneta admin theme is getting one step away from in context editing with presenting the admin interface in a consistent and non polluted default Plone style. Which is awesome!

I was exploring the next step away from traditional Plone in context editing: An admin interface where once you are in admin mode you never jump to a view except when you explicitly choose to. I think this will make editing quicker and easier.

It is just an little experiment and not a feature request. I was just exploring what it takes to accomplish ;)

**Comment by davilima6** [comment]
Actually it's an indication that the link _opens in new window_ rather than an _external_ link.

Indeed as much as I see the need for informing users of that behaviour I agree it should be an "eye" icon for consistency and increased learnability. We could have a `title` on it informing user about target=_blank behaviour. Then eye, pencil and lightning icons would all live inside `Actions` column. 

Re: detours, I also agree it's too easy and a bit strange to leave _file-explorer mode_ once you click a non-folderish item. However I wouldn't redirect to Edit mode as you proposed. I'd leave it as is because it's expected and fundamental Plone behaviour not to get into Edit screens without explicit Edit action. I think that's based on the belief it's less expensive and less riskier to get to `View` view by accident than to `Edit` view. Maybe you could build a plugin to provide a configurable manner for each site to decide preferred redirection behaviour. If many people use it we can include it in Plone core.
